sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import java.util.Objects;
+
+/**
+ * It represents a single header for a request being sent to Azure Service.
+ *
+ * If multiple header values are added to a request or response with the same name (case-insensitive), then the values
+ * will be appended at the end of the same {@link Header} with commas separating them.
+ */
+public class Header {
+    private final String name;
+    private String value;
+
+    /**
+     * Create a Header instance using the provided name and value.
+     *
+     * @param name the name
+     * @param value the value
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public Header(String name, String value) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.value = value;
+    }
+
+    /**
+     * Gets the header name.
+     *
+     * @return the name of this Header
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the header value.
+     *
+     * @return the value of this Header
+     */
+    public String getValue() {
+        return value;
+    }
+
+    /**
+     * Gets the comma separated value as an array.
+     *
+     * @return the values of this Header that are separated by a comma
+     */
+    public String[] getValues() {
+        return value == null ? null : value.split(",");
```

Comment:
We should see if it is cheaper to store the values as a `String[]`, `List<String>`, or `String`. Not required for this PR.

Reply:
This will be moved to a follow-up issue.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/15219